OpenRA.Game/Map/Map.cs

```diff
@@ -804,6 +804,17 @@ public CPos ChooseRandomCell(MersenneTwister rand)
 			return new MPos(x, y).ToCPos(this);
 		}
 
+		public CPos ChooseClosestEdgeCell(CPos pos)
+		{
+			var x = pos.X < (Bounds.Right / 2) ? 0 : Bounds.Right;
+			var y = pos.Y < (Bounds.Bottom / 2) ? 0 : Bounds.Bottom;
```

Comment:
Bounds are defined in map coords. You need to do your checks in those, and then convert to CPos at the end.